macaroonbakery/macaroon.py

```diff
@@ -113,51 +115,39 @@ def add_caveat(self, cav, key=None, loc=None):
             id = self._new_caveat_id(self.caveat_id_prefix)
             self.caveat_data[id] = caveat_info
 
-        m = self._macaroon.add_third_party_caveat(cav.location, root_key, id)
-        new_macaroon = copy.copy(self)
-        new_macaroon._macaroon = m
-        return new_macaroon
+        self._macaroon.add_third_party_caveat(cav.location, root_key, id)
 
     def add_caveats(self, cavs, key, loc):
-        '''Return a new macaroon with all caveats added.
+        '''Add all caveats to the macaroon.
 
         This method does not mutate the current object.
         @param cavs arrary of caveats.
         @param key the nacl public key to encrypt third party caveat.
         @param loc locator to find the location object that has a method
         third_party_info.
-        @return a new macaroon object with the given caveats.
         '''
-        macaroon = self
+        if cavs is None:
+            return
         for cav in cavs:
-            macaroon = macaroon.add_caveat(cav, key, loc)
-        return macaroon
+            self.add_caveat(cav, key, loc)
 
     def serialize(self):
-        '''Return a dictionary holding the macaroon data in V1 JSON format.
+        '''Return a dictionary holding the macaroon data in JSON format.
```

Comment:
I wonder if this method should be named serialize_json to indicate that it's a JSON serialization method - the default behaviour of the pymacaroons serialization is to serialize as binary and it might be surprising to that this uses JSON,